packages/helpers/classes/personalization.js

```diff
@@ -271,6 +274,34 @@ class Personalization {
     this.substitutionWrappers = wrappers;
   }
 
+  /**
+   * Reverse merge dynamic template data, preserving existing ones
+   */
+  reverseMergeDynamicTemplateData(dynamicTemplateData) {
```

Comment:
Can we call this method `deepMergeDynamicTemplateData` instead? I think that's what's happening here, so that description would be a bit clearer and a better fit imo.